cmake/utils.cmake

```diff
@@ -145,3 +145,16 @@ function(list_add_prefix
     endforeach()
     set(${list_out} ${result} PARENT_SCOPE)
 endfunction()
+
+# string() has JOIN option but only since 3.12.
+function(string_join
+    glue
+    string_out
+)
+    set(result "")
+    foreach(item IN LISTS ARGN)
+        set(result "${result}${item}${glue}")
+        message(STATUS "result=${result}")
```

Comment:
The `message` is obviously redundant.

In contrast to the builtin function, this one adds a dangling glue to the end of the result string.

Reply:
Fixed.